presto-spi/src/main/java/io/prestosql/spi/connector/classloader/ForClassLoaderSafe.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.spi.connector.classloader;
+
+import javax.inject.Qualifier;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target({FIELD, PARAMETER, METHOD})
+@Qualifier
+public @interface ForClassLoaderSafe
```

Comment:
I'm thinking we should drop the `For` here. I can sort of see why we started it, but I'm not sure it provides any value here. @martint thoughts?

Reply:
I like `For`. It clearly says that is dedicated for guice. Also it mention that given component is going to be used by something. Without `For` it would say that given component has such "property".

```
binder.bind(ConnectorSplitManager.class)
    .annotatedWith(ForClassLoaderSafe.class)
    .to(KafkaSplitManager.class)
    .in(Scopes.SINGLETON);
```
vs

```
binder.bind(ConnectorSplitManager.class)
    .annotatedWith(ClassLoaderSafe.class)
    .to(KafkaSplitManager.class)
    .in(Scopes.SINGLETON);
```

First example clearly states that `KafkaSplitManager` will be consumed by `ForClassLoaderSafe` component. While the second example misleads saying that `KafkaSplitManager` is  `ClassLoaderSafe`.